SaraAlert/SaraAlert
# URGENT BUGFIX: Adding assessments broken for non-US mainland users

# Description
Jira Ticket: [SARAALERT-400](https://tracker.codev.mitre.org/browse/SARAALERT-400)

Bugfix for #1119 ([slack context](https://mitrecorp.slack.com/archives/G01792S9Z5M/p1636033833019500)) and also correct calculated symptom onset logic to be based on `reported_at` rather than `created_at`.

The way the time picker was parsing input caused an error. The moment library can only parse certain time zone abbreviations (mostly US mainland time zones and GMT), so when users outside of those zones, e.g. Guam, were trying to create or edit reports, an error was thrown from the date picker.

This fixes it by removing the time zone from the date picker input entirely. Instead, we just display the time zone abbreviation next to the input.

This also removes some unnecessary conversions we were doing between JS date and moment.

## Demo/Screenshots
<img width="495" alt="image" src="https://user-images.githubusercontent.com/8235974/140440495-9cd7b38c-6e88-461b-9a4c-a7dcb6caa8fb.png">

## How to Replicate
- Change you computer's time zone to something other than US or GMT (other than what's documented [here](https://momentjs.com/docs/#/parsing/string/) (scroll down a little, it uses RFC 2822))
- Go to a patient page and click the Add New Report button

## Solution
Described above and in commit message

## Important Changes
`SymptomAssessment.js`
- Add new time zone label

`DateInput.js`
- Remove unnecessary conversions between JavaScript date and moment.

`DateTimeInput.js`
- Remove the time zone from the input
- Note the width change here is the narrowest the input can be and still accommodate the widest possible value ("08/08/2020 08:08")

## Testing Recommendations
- Try from a couple different time zones, including US and non-US time zones
- Add new report
- Edit existing report
- Modify the date and time of the report in different ways, including clicking on the date picker and using the cursor and keyboard to edit the value directly
- Click submit and ensure it saves and that the resulting date, time, and zone are correct
- Ensure the history item for saving a reported at change still works as it should and includes the right time zone
- Clear symptom onset and then create a symptomatic report and back date it. Ensure the calculated symptom onset is correct. Note that there could be a diff by one day if you're using a different time zone. The calculated symptom onset uses the *patient's* time zone and report is entered with the *user's* time zone. Correcting this would be out of the scope of the ticket, as it's always been an issue. Worth noting most of the time, the user's time zone will be the same as the patient.

# Checklists

**Submitter:**


@ngfreiter:


@sgober:


@timwongj:
